### PR TITLE
[Feature] implement iceberg v2 read for orc

### PR DESCRIPTION
**Pull request by cbcbq**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

implementation for iceberg v2 table reading of orc-format
related with: [#3452]

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by DorianZheng** [comment]
run starrocks_be_unittest

**Comment by DorianZheng** [comment]
run starrocks_be_unittest

**Comment by DorianZheng** [comment]
run starrocks_admit_test

File: be/src/exec/vectorized/iceberg/iceberg_delete_builder.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
```
Comment (DorianZheng) [inline be/src/exec/vectorized/iceberg/iceberg_delete_builder.h:1]: we have changed license to Apache License

File: be/src/exec/vectorized/hdfs_scanner_orc.cpp [hunk 2]
```diff
@@ -477,6 +488,15 @@ Status HdfsOrcScanner::do_init(RuntimeState* runtime_state, const HdfsScannerPar
     _should_skip_file = false;
     _use_orc_sargs = true;
     // todo: build predicate hook and ranges hook.
+    if (!scanner_params.deletes.empty()) {
```
Comment (DorianZheng) [inline be/src/exec/vectorized/hdfs_scanner_orc.cpp:491]: add stats to monitor builder elapse time

**Review by DorianZheng — COMMENTED** [review]
left comments, rest LGTM

File: be/src/exec/vectorized/iceberg/iceberg_delete_builder.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+static const IcebergColumnMeta k_delete_file_pos{
+        .id = INT32_MAX - 102, .col_name = "pos", .type = TPrimitiveType::BIGINT};
+
+Status ORCPositionDeleteBuilder::build(const std::string& timezone, const std::string& delete_file_path,
+                                       int64_t file_length, std::set<int64_t>* need_skip_rowids) {
+    std::vector<SlotDescriptor*> slot_descriptors{&(IcebergDeleteFileMeta::get_delete_file_path_slot()),
+                                                  &(IcebergDeleteFileMeta::get_delete_file_pos_slot())};
+
+    std::unique_ptr<RandomAccessFile> file;
+    ASSIGN_OR_RETURN(file, _fs->new_random_access_file(delete_file_path));
```
Comment (DorianZheng) [inline be/src/exec/vectorized/iceberg/iceberg_delete_builder.cpp:30]: could leverage file cache, refers to `HdfsScanner::open_random_access_file`

**Review by dirtysalt — DISMISSED** [review]

File: be/src/exec/vectorized/iceberg/iceberg_delete_builder.cpp [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            auto str = strings::Substitute("delete file schema doesn't meet requirement, need: [file_path, pos]");
+            LOG(WARNING) << str;
+            return Status::InternalError(str);
+        }
+
+        auto* file_path_col = static_cast<BinaryColumn*>(chunk->get_column_by_slot_id(k_delete_file_path.id).get());
+        auto* position_col = static_cast<Int64Column*>(chunk->get_column_by_slot_id(k_delete_file_pos.id).get());
+        for (auto row = 0; row < chunk_size; row++) {
+            if (file_path_col->get_slice(row) != _datafile_path) {
+                LOG(WARNING) << "read path not matched, read " << file_path_col->get_slice(row) << " expect "
```
Comment (DorianZheng) [inline be/src/exec/vectorized/iceberg/iceberg_delete_builder.cpp:111]: this should be a debug level log

**Review by DorianZheng — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

**Comment by cbcbq** [comment]
run starrocks_admit_test

**Review by DorianZheng — DISMISSED** [review]

**Review by Youngwb — APPROVED** [review]

File: be/src/exec/hdfs_scanner.h [hunk 5]
```diff
@@ -297,6 +302,7 @@ class HdfsScanner {
     // by default it's no compression.
     CompressionTypePB _compression_type = CompressionTypePB::NO_COMPRESSION;
     std::shared_ptr<io::CacheInputStream> _cache_input_stream = nullptr;
+    std::set<std::int64_t> _need_skip_rowids;
```
Comment (stephen-shelby) [inline be/src/exec/hdfs_scanner.h:305]: nit: use bitmap like BitmapPositionDeleteIndex

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**